src/compiler/checker.ts

```diff
@@ -10403,6 +10433,7 @@ namespace ts {
                 case SyntaxKind.TemplateExpression:
                     return checkTemplateExpression(<TemplateExpression>node);
                 case SyntaxKind.StringLiteral:
+                    return checkStringLiteralExpression(<LiteralExpression>node);
                 case SyntaxKind.NoSubstitutionTemplateLiteral:

```

Comment:
@tinganho In case you're curious about your initial string comparison benchmark: the VM is comparing the same pointer to the strings, see here for details on string comparisons:
http://blog.cdleary.com/2012/01/string-representation-in-spidermonkey/